torch/nn/modules/module.py

```diff
@@ -320,32 +323,46 @@ def register_forward_hook(self, hook):
         return handle
 
     def __call__(self, *input, **kwargs):
-        for hook in self._forward_pre_hooks.values():
-            hook(self, input)
-        result = self.forward(*input, **kwargs)
-        for hook in self._forward_hooks.values():
-            hook_result = hook(self, input, result)
-            if hook_result is not None:
-                raise RuntimeError(
-                    "forward hooks should never return any values, but '{}'"
-                    "didn't return None".format(hook))
-        if len(self._backward_hooks) > 0:
-            var = result
-            while not isinstance(var, Variable):
-                if isinstance(var, dict):
-                    var = next((v for v in var.values() if isinstance(v, Variable)))
-                else:
-                    var = var[0]
-            grad_fn = var.grad_fn
-            if grad_fn is not None:
-                for hook in self._backward_hooks.values():
-                    wrapper = functools.partial(hook, self)
-                    functools.update_wrapper(wrapper, hook)
-                    grad_fn.register_hook(wrapper)
+        tracing_state = None
+        if torch.jit._tracing:
+            input_vars = tuple(torch.autograd.function._iter_variables(input))
```

Comment:
This is worrying, because this means that we won't be able to trace models that take arbitrary things as inputs (`_iter_variables` has some assumptions about types - they should be nested tuples, lists and dicts of Variables)

Reply:
I see. I use `input_vars` to obtain `tracing_state` with `torch.jit.get_tracing_state`, which is only interested in Variables, if I got it right.
I don't have other ideas except accessing the tracing state from somewhere else...